iroha_actor/src/broker.rs

```diff
@@ -169,62 +84,87 @@ impl Default for Broker {
 impl Broker {
     /// Default constructor for broker
     pub fn new() -> Self {
-        Self(Arc::new(DashMap::new()))
+        Self(Arc::new(MessageMap::new()))
     }
 
-    /// Send message via broker
-    pub async fn issue_send<M: BrokerMessage + Send + Sync>(&self, message: M) {
-        let warn_empty = || {
-            iroha_logger::warn!(
-                "The message {:?} was send but no entity is subscribed to it.",
-                TypeId::of::<M>()
-            )
+    fn entry(&'_ self, id: TypeId) -> Entry<'_, TypeId, ActorMap<BrokerRecipient>> {
+        self.0.entry(id)
+    }
+
+    /// Number of subscribers for specific message
+    pub fn subscribers<M: BrokerMessage + Send + Sync>(&self) -> usize {
+        let entry = if let Entry::Occupied(entry) = self.entry(TypeId::of::<M>()) {
+            entry
+        } else {
+            return 0;
         };
-        if let Some(subscribers) = self.0.get(&TypeId::of::<M>()) {
-            if subscribers.is_empty() {
-                warn_empty();
-            } else {
-                subscribers.value().publish_message(message).await;
-            }
+
+        entry
+            .get()
+            .iter()
+            .filter_map(|(id, recipient)| Some((id, recipient.downcast_ref::<Recipient<M>>()?)))
+            .fold(0, |p, (_, n)| p + if n.0.is_closed() { 0 } else { 1 })
+    }
+
+    /// Send message via broker
+    #[allow(clippy::missing_panics_doc)]
+    pub async fn issue_send<M: BrokerMessage + Send + Sync>(&self, m: M) {
+        let mut entry = if let Entry::Occupied(entry) = self.entry(TypeId::of::<M>()) {
+            entry
         } else {
-            warn_empty();
+            return;
+        };
+        let send = entry
+            .get()
+            .iter()
+            .filter_map(|(id, recipient)| Some((id, recipient.downcast_ref::<Recipient<M>>()?)))
+            .map(|(id, recipient)| {
+                let m = m.clone();
+                async move {
+                    if recipient.0.is_closed() {
+                        return Some(*id);
+                    }
+
+                    recipient.send(m).await;
+                    None
+                }
+            });
+
+        let closed: Vec<_> = future::join_all(send).await;
```

Comment:
If you noticed in my PR I used futures unordered. Because as `join_all` is simple and inefficient - check this issue https://github.com/tokio-rs/tokio/issues/2401#issuecomment-612858572 and also please check FuturesUnordered and join_all docs. They all point it out.

As everything is based on actors in our code - performance is crucial here.